app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -47,11 +54,32 @@ class MessageCardViewHolder(
     @Composable
     override fun Content() {
         val message by remember { mutableStateOf(messageGlobal) }
+        val wallpaperState = components.appStore
+            .observeAsComposableState { state -> state.wallpaperState }.value ?: WallpaperState.default
+        val isWallpaperNotDefault = !Wallpaper.nameIsDefault(wallpaperState.currentWallpaper.name)
+        val defaultMessageCardColors = MessageCardColors.buildMessageCardColors()
+
+        var buttonColor = defaultMessageCardColors.buttonColor
+        var buttonTextColor = defaultMessageCardColors.buttonTextColor
```

Comment:
```suggestion
        var (buttonColor, buttonTextColor) = MessageCardColors.buildMessageCardColors()
```
Wondering if we could use destructuring declaration here to save us a few lines https://kotlinlang.org/docs/destructuring-declarations.html

Reply:
If we wanted to go fully immutable you may be able to do something like this, even:

val (buttonColor, buttonTextColor) = if (isWallpaperNotDefault) {
    FirefoxTheme.colors.layer1 to if (!isSystemInDarkTheme()) { FirefoxTheme.colors.textActionSecondary } else defaultMessageCardColors.buttonTextColor
} else {
   MessageCardColors.buildMessageCardColors()
}

Reply:
I had no idea about the destructuring! that's really cool! I had to just insert some underscores though, as the destructuring is tied to the ordering of the parameters.

```
var (_, _, _, _, buttonColor, buttonTextColor) = MessageCardColors.buildMessageCardColors()
```